Assets/Scripts/EnemyTank/EnemyTankController.cs

```diff
@@ -4,35 +4,48 @@
 
 public class EnemyTankController {
 
-    private TankModel tankModel;
-    private EnemyTankView tankView;
+    public TankModel tankModel;
+    public EnemyTankView tankView;
     public EnemyTankService tankService;
 
-    public Rigidbody rgbd { private get; set; }
+    public Rigidbody rgbd { get; set; }
 
-    int target;
-    List<Vector3> waypoints;
+    private EnemyTankState currentState;
 
-    public EnemyTankController(TankModel model, EnemyTankView view, List<Vector3> points, int to) {
+
+    public EnemyTankController(TankModel model, EnemyTankView view, Vector3 p1, Vector3 p2) {
         tankModel = model;
         tankView = view;
 
         tankView.SetTankController(this);
 
-        waypoints = points;
-        target = (to + 1) % waypoints.Count;
+        ChangeState(tankView.GetComponent<PatrolState>());
```

Comment:
Avoid Multiple GetComponent calls since they use string comparison. Store the reference , then use it.